yew/src/virtual_dom/vlist.rs

```diff
@@ -172,6 +439,950 @@ mod tests {
             </div>
         };
     }
+
+    fn new_keyed_vtag<T: Into<Cow<'static, str>>, K: ToString>(tag: T, key: K) -> VTag {
+        let mut vtag = VTag::new(tag.into());
+        vtag.key = Some(Key::from(key.to_string()));
+        vtag
+    }
+
+    #[cfg(feature = "web_sys")]
+    fn inner_html(element: &Element) -> String {
+        element.inner_html()
+    }
+
+    #[cfg(feature = "std_web")]
+    fn inner_html(element: &Element) -> String {
+        use stdweb::unstable::TryInto;
+        use stdweb::web::{HtmlElement, IHtmlElement};
+        let html_element: HtmlElement = element
+            .clone()
+            .try_into()
+            .expect("Failed to convert element into HtmlElement");
+        html_element.inner_text()
+    }
+
+    #[test]
+    fn vlist_vdiff_apply_non_keyed_from_none_works_with_all_vnode_types_as_children() {
+        let vchild: VChild<Comp> = VChild::new(
+            CountingCompProps {
+                id: 0,
+                panic_if_changes: false,
+            },
+            NodeRef::default(),
+            None,
+        );
+        let vref_element: Element = crate::utils::document().create_element("i").unwrap();
+        let vref_node: Node = vref_element.clone().into();
+        let mut vlist = VList::new_with_children(
+            vec![
+                VNode::VText(VText::new("a".into())),
+                VNode::VTag(Box::new(VTag::new("span"))),
+                VNode::VText(VText::new("c".into())),
+                VNode::VText(VText::new("d".into())),
+                VNode::VComp(vchild.into()),
+                VNode::VList(VList::new_with_children(
+                    vec![
+                        VNode::VText(VText::new("foo".into())),
+                        VNode::VText(VText::new("bar".into())),
+                    ],
+                    None,
+                )),
+                VNode::VRef(vref_node),
+            ],
+            None,
+        );
+
+        let parent_scope: AnyScope = Scope::<Comp>::new(None).into();
+        let parent_element = crate::utils::document().create_element("div").unwrap();
+        vlist.apply(&parent_scope, &parent_element, NodeRef::default(), None);
+
+        assert_eq!(
+            inner_html(&parent_element),
+            "a<span></span>cd<p>0</p>foobar<i></i>",
+            "The VList didn't render properly."
+        );
+    }
+
+    #[test]
+    fn vlist_vdiff_apply_keyed_from_none_works_with_all_vnode_types_as_children() {
```

Comment:
Can you move all of the keyed tests into the `layout_tests` module? I think you could follow a similar model to how the `diff()` test works, maybe `diff_keyed()`?